tasks/frontEnd/src/components/ReviewForm.vue

```diff
@@ -0,0 +1,138 @@
+    </DropZone>
+    <Divider />
+    <Button type="submit" :disabled="disableSubmit">Submit</Button>
+  </Form>
+</template>
+
+<script setup>
+import { reactive, ref, watch } from 'vue';
+
+const emit = defineEmits(['addReview'])
+
+const review = reactive({ shopName: '', productName: '', userName: '', userEmail: '', rating: 3, reviewBody: '', verified: false, files: [] });
+
+const shopNameError = ref(null);
+const productNameError = ref(null);
+const userNameError = ref(null);
+const userEmailError = ref(null);
+const reviewBodyError = ref(null);
+
+const disableSubmit = ref(true);
+
+watch(review, () => {
+  if(shopNameError.value == '' && productNameError.value == '' && userNameError.value == '' && userEmailError.value == '' && reviewBodyError.value == '') {
+    disableSubmit.value = false;
+  } else {
+    disableSubmit.value = true;
+  }
+});
+
+watch(() => review.shopName, () => {
+  if(review.shopName.length == 0) {
+    shopNameError.value = 'This is a required field';
+  } else {
+    shopNameError.value = '';
+  }
+});
+
+watch(() => review.productName, () => {
+  if(review.productName.length == 0) {
+    productNameError.value = 'This is a required field'
+  } else {
+    productNameError.value = '';
+  }
+});
+
+watch(() => review.userName, () => {
+  if(review.userName.length == 0) {
+    userNameError.value = 'This is a required field'
+  } else {
+    userNameError.value = '';
+  }
+});
+
+watch(() => review.userEmail, () => {
+  if(review.userEmail.length == 0 || !(/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(review.userEmail))) {
+    userEmailError.value = 'Please provide a valid email';
+  } else {
+    userEmailError.value = '';
+  }
+});
+
+watch(() => review.reviewBody, () => {
+  if(review.reviewBody.length == 0) {
+    reviewBodyError.value = 'This is a required field';
+  } else {
+    reviewBodyError.value = '';
+  }
+});
+
+const handleSubmit = () => {
+  let mockImages = [];
+  for (let file in review.files) {
+    mockImages = [...mockImages, 'https://loremflickr.com/640/480'];
+  }
+  const newReview = { 
+    shopName: review.shopName, 
+    productName: review.productName, 
+    reviewer: { 
+      name: review.userName,
+      email: review.userEmail
+    },
+    review: { 
+      rating: review.rating,
+      reviewBody: review.reviewBody,
+      reviewImages: mockImages, 
+      verifiedPurchase: review.verified
+    }
+  };
+  const requestOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(newReview)
+  };
+  fetch('https://663e3cbae1913c47679705a2.mockapi.io/api/v1/reviews', requestOptions )
+    .then((res) => {
+      if (res.ok) {
+        emit('addReview');
+      }
+    });
+}
+
+const handleDrop = (_dropFiles, acceptedFiles, _rejectedFiles) => {
+  review.files = [...review.files, ...acceptedFiles];
+};
+
+const getSource = (file) => {
+  return window.URL.createObjectURL(file);
```

Comment:
This should be revoked using `window.URL.revokeObjectURL` to avoid memory leaks, and release the object when we are done using it

Reply:
Good note, this was actually taken from the DropZone documentation on PolarisVue so I didn't give it too much of a critical eye